File: orchestrator/src/main/java/br/com/orchestrator/application/dto/Event.java
package br.com.orchestrator.application.dto;

import br.com.orchestrator.core.enums.EEventSource;
import br.com.orchestrator.core.enums.ESagaStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

import static org.springframework.util.ObjectUtils.isEmpty;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Event {

    private String id;
    private String userId;
    private String transactionId;
    private User payload;
    private EEventSource source;
    private ESagaStatus status;
    private List<History> eventHistory;
    private LocalDateTime createdAt;

    public void addToHistory(History history) {
        if (isEmpty(eventHistory)) {
            eventHistory = new ArrayList<>();
        }
        eventHistory.add(history);
    }
}

File: validated-document-service/src/main/kotlin/br/com/validateddocumentservice/application/dto/Event.kt
package br.com.validateddocumentservice.application.dto

import br.com.validateddocumentservice.core.enums.ESagaStatus
import lombok.AllArgsConstructor
import lombok.Builder
import lombok.Data
import lombok.NoArgsConstructor
import org.springframework.util.ObjectUtils
import java.time.LocalDateTime


class Event {
    private val id: String? = null
    private val userId: String? = null
    private val transactionId: String? = null
    private val payload: User? = null
    private var source: String? = null
    private var status: ESagaStatus? = null
    private var eventHistory: MutableList<History>? = null
    private val createdAt: LocalDateTime? = null

    fun addToHistory(history: History) {
        if (ObjectUtils.isEmpty(eventHistory)) {
            eventHistory = ArrayList()
        }
        eventHistory!!.add(history)
    }

    fun getId(): String? {return this.id}
    fun getUserId(): String? { return this.userId};
    fun getTransactionId(): String? { return this.transactionId};
    fun getPayload(): User? { return this.payload};
    fun getSource(): String? { return this.source};
    fun getStatus(): ESagaStatus? { return this.status};
    fun setStatus(status: ESagaStatus) { this.status = status};
    fun setSource(source: String) { this.source = source}

    override fun toString(): String {
        return "Event(id=$id, userId=$userId, transactionId=$transactionId, payload=$payload, source=$source, status=$status, eventHistory=$eventHistory, createdAt=$createdAt)"
    };
}